cmd/soroban-cli/src/commands/contract/deploy/asset.rs

```diff
@@ -108,8 +110,14 @@ impl NetworkRunnable for Cmd {
             network_passphrase,
             &key,
         )?;
+        if self.fee.build_only {
+            return Ok(TxnResult::Txn(tx));
+        }
         let txn = client.create_assembled_transaction(&tx).await?;
         let txn = self.fee.apply_to_assembled_txn(txn);
+        if self.fee.sim_only {
+            return Ok(TxnResult::Txn(txn.transaction().clone()));
+        }
```

Comment:
I pushed a change moving the `if sim_only` check to the same level as the `if build_only` check. If you look above at the few lines captured on this comment, the sameness and likeness of the two (I think) make it really easy to understand the story they form.

The other reason I suggest this change is because the `apply_to_assembled_txn` was ambiguous when it returned a `Transaction`/XDR String or an `Assembled`. It's difficult to know what state that tx is in, and why it's being passed back separately from the Assembled, given that the Assembled actually contains the Transaction anyway. This way we let each command on its own control their flow.